### PR TITLE
Refactor template take3

### PR DESCRIPTION
**Pull request by maartenbreddels**
This replaces #30 

PS: I did a rename + edit in different commits, but this PR still shows the removal of a file, and a new file (gridstack_base.tpl -> gridstack_base.html.j2).

TODO: 
 * [ ] check if output is still the same
 * [ ] apply test corrections of #30 
 * [ ] think of a way to configure the option of showing the handles with nbconvert

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
@maartenbreddels I release Voila 0.2.0a0, which you should be able to use for this PR.

File: setup.py [hunk 1]
```diff
@@ -1,5 +1,78 @@
...
+
+
+# these are based on jupyter_core.paths
+def jupyter_config_dir():
+    """Get the Jupyter config directory for this platform and user.
+    Returns JUPYTER_CONFIG_DIR if defined, else ~/.jupyter
+    """
+
+    env = os.environ
+    home_dir = get_home_dir()
```
Comment (SylvainCorlay) [inline setup.py:17]: if seems that get_home_dir is not defined.

File: share/jupyter/nbconvert/templates/gridstack/grid.html.j2.orig [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    </div>
+    <div id="gsConfig" style="visibility: hidden">
+    </div>
+</div>
+{% endif %}
+{% endblock body %}
+=======
+{{ super() }}
+{% endblock body_footer %}
+>>>>>>> refactor: use new template system of voila/nbconvert:share/jupyter/nbconvert/templates/gridstack/grid.html.j2
```
Comment (maartenbreddels) [inline share/jupyter/nbconvert/templates/gridstack/grid.html.j2.orig:190]: I'm assuming this file was accidently commited, and I'll remove it.

**Comment by maartenbreddels** [comment]
If I run with `$ voila --template gridstack ~/src/QuantStack/voila/notebooks/basics.ipynb --VoilaConfiguration.resources='{"gridstack": {"show_handles": True}}'` I cannot reach the bottom right handle, it disappears before I can get there with the mouse cursor.

**Comment by maartenbreddels** [comment]
This is also the case in master, and seems to be related to the config, introduced in #37 (cc @mariobuikhuizen )

**Comment by SylvainCorlay** [comment]
Thanks!

This looks good. I think I need to tag a release targeting 0.1 now, and then I will merge and move forward.

File: setup.py [hunk 3]
```diff
@@ -1,5 +1,44 @@
...
+
+        for prefix_target, name in self.prefix_targets:
+            source = os.path.join('share', 'jupyter', prefix_target, name)
+            target = os.path.join(target_dir, prefix_target, name)
+            target_subdir = os.path.dirname(target)
+            if not os.path.exists(target_subdir):
+                os.makedirs(target_subdir)
+            rel_source = os.path.relpath(os.path.abspath(source), os.path.abspath(target_subdir))
+            try:
+                os.remove(target)
```
Comment (martinRenou) [inline setup.py:34]: Shouldn't we let `os.symlink` fail instead of removing the target?

Comment (maartenbreddels) [reply setup.py:34]: it usually means there is an old directory or old symlink, effectively overwrite behavior.

**Pull request merged**